src/formatting/imports.rs

```diff
@@ -560,6 +574,7 @@ impl UseTree {
                 SharedPrefix::Module => {
                     self.path[..self.path.len() - 1] == other.path[..other.path.len() - 1]
                 }
+                SharedPrefix::One => unreachable!(),
```

Comment:
Did you try making this branch return `true`? It seems like that could mean you don't need `temp_merge_by`.

Reply:
Yes, I tried that, but I found that ended up requiring additional special logic to address `SharedPrefix::One` cases.
For example, consider the following input:

```rust
use b;
use a::ac::{aca, acb};
use a::{aa::*, ab};
```

If we changed `SharedPrefix::One => unreachable!(),` to `SharedPrefix::One => true,` with the rest code reverted to `master`, the above code would be formatted into:

```rust
use {
    a::{
        ab,
        ac::acb,
        {aa::*, ac::aca},
    },
    b,
};
```

Here, `ac::acb` and `ac::aca` should be put together, but actually not.
After bumping into this issue, I thought it would be a good and simple solution to leverage the same logic as the `SharedPrefix::Crate` and then to merge the result.

Reply:
Thanks @msmorgan for reviewing and @magurotuna for explaining the test case. I'm not sure about the implementation to deal with those types of cases though. I suspect the `unreachable` arm and temp/actual vars will be puzzling to both casual and future browsers of the code.

I will take a closer look at the proposed changes later on, but I think it will be preferable if we can come up with an approach that avoids those

Reply:
Okay, I will consider more as well.

Reply:
@calebcartwright 
I attempted to fix the code to avoid `unreachable!` arm. I think this implementation is kind of naive but seems to work fine. To ensure it works, I added more test cases.
Could you take a look again?

Reply:
> I think this implementation is kind of naive but seems to work fine

As in, you think your original approach was better than this, or you think there's a better approach than this updated implementation?

> To ensure it works, I added more test cases

Brilliant thank you! If you don't already have these, I'd advising having a few cases that exercise some `self` and comment scenarios

> Could you take a look again?

Yes indeed. I've not had a ton of bandwidth to allocate to PR reviews lately though so may be a while yet.


Also, don't necessarily feel like you have to shoehorn the One strategy into the existing flow that's leveraged for other variants, especially if doing so is more cumbersome than helpful.

Reply:
> As in, you think your original approach was better than this, or you think there's a better approach than this updated implementation?

Actually, on second thought, now I think the current approach is reasonably good. Although there are a couple of ad-hoc if-expressions, which seemed a little bit ugly to me, it apparently works just fine.  Implementing imports_granularity essentially involves complexity so I feel like working fine is more important than anything else. 

> If you don't already have these, I'd advising having a few cases that exercise some self and comment scenarios

Those definitely should be added! I've added more tests that cover them.